src/Websockettest.js

```diff
@@ -483,14 +499,17 @@ const RMBTTest = (function() {
         //and max MAX_CHUNKSIZE
         calculatedChunkSize = Math.min(MAX_CHUNK_SIZE, calculatedChunkSize);
 
-        debug("calculated chunksize for upcoming speed test " + calculatedChunkSize / 1024 + " KB");
+        logger.debug("calculated chunksize for upcoming speed test " + calculatedChunkSize / 1024 + " KB");
 
         if (limitToExistingChunks) {
             //get closest chunk size where there are saved chunks available
             let closest = Number.POSITIVE_INFINITY;
             Object.keys(_arrayBuffers).forEach((key) => {
                 let diff = Math.abs(calculatedChunkSize - key);
                 if (diff < Math.abs(calculatedChunkSize - closest)) {
+                    if (typeof key == "string") {
```

Comment:
during testing I ran into problems where the key sometimes was a string for some reason. I could not find the source of the problem and was only able to create this temporary fix.
I will investigate later some more. (Added comment)